applications/StructuralMechanicsApplication/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_base_condition.h

```diff
@@ -62,23 +63,25 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION)  AdjointSemiAnalyticBaseCondi
     ///@name Life Cycle
     ///@{
 
-    /// Default constructor.
-    AdjointSemiAnalyticBaseCondition(): Condition()
+    AdjointSemiAnalyticBaseCondition(IndexType NewId = 0)
+    : Condition(NewId), mPrimalCondition(NewId, pGetGeometry())
     {
     }
 
-    AdjointSemiAnalyticBaseCondition(
-        Condition::Pointer pPrimalCondition
-        ): Condition( pPrimalCondition->Id(), pPrimalCondition->pGetGeometry(), pPrimalCondition->pGetProperties() ),
-           mpPrimalCondition(pPrimalCondition)
-        {
-        }
+    AdjointSemiAnalyticBaseCondition(IndexType NewId, GeometryType::Pointer pGeometry)
+    : Condition(NewId, pGeometry), mPrimalCondition(NewId, pGeometry)
+    {
+    }
 
-    /// Destructor.
-    ~AdjointSemiAnalyticBaseCondition() override
+    AdjointSemiAnalyticBaseCondition(IndexType NewId,
+                        GeometryType::Pointer pGeometry,
+                        PropertiesType::Pointer pProperties)
+    : Condition(NewId, pGeometry, pProperties), mPrimalCondition(NewId, pGeometry, pProperties)
     {
     }
 
+    virtual ~AdjointSemiAnalyticBaseCondition() {}
```

Comment:
```suggestion
    ~AdjointSemiAnalyticBaseCondition() override = default;
```

not even sure if you need a destructor at all, since the compiler creates a default one...
The things I am not 100% sure abt is how it works with the inheritance

Reply:
From what i understand from this 

https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors

there should be a virtual destructor for any class that may be overridden. `override = default` is something i can add though.

Reply:
you are right in the sense that you have to add it to BASEclasses
I am still not sure if it makes sense to define a destructor for derived classes if the base alseady has a virtual one

Reply:
I will check again for this intermediate level of inheritance.

Reply:
same, you were right